tenets/codelingo/effective-go/loop-variable-used-in-go-routine/test.go

```diff
@@ -0,0 +1,79 @@
+package main
+
+import (
+	"fmt"
+)
+
+func Test1(values []int) {
+	for val := range values {
+		go func() { //ISSUE
+			fmt.Println(val)
+		}()
+	}
+}
+
+func Test2(values []int) {
+	for val := range values {
+		go func(val int) {
+			fmt.Println(val)
+		}(val)
+	}
+}
+
+func Test3(values []int, n int) {
+	for i, val := range values {
+		go func(val int) { //ISSUE
+			if i < n {
+				fmt.Println(val)
+			}
+		}(val)
+	}
+}
+
+func Test4(values []int, n int) {
+	for i, val := range values {
+		fmt.Println(i, val)
+		go func() {
+			fmt.Println(n)
+		}()
+	}
+}
+
+func Test5(values []int, n bool) {
+	for val := range values {
+		go func(val int, n bool) {
+			if n {
+				fmt.Println(val)
+			}
+		}(val, n)
+	}
+}
+
+func Test6(values []int, n bool) {
+	for val := range values {
+		go func(n bool) { //ISSUE
```

Comment:
What's wrong with this one?

Reply:
Because `val` is used in the goroutine and not passed.

Reply:
Ah right. I thought the comment was saying the issue was with the bool.